src/main/java/com/radixdlt/engine/RadixEngine.java

```diff
@@ -42,27 +42,27 @@
 	private final ConstraintMachine constraintMachine;
 	private final CMStore virtualizedCMStore;
 	private final EngineStore<T> engineStore;
-	private final CopyOnWriteArrayList<AtomEventListener<T>> atomEventListeners = new CopyOnWriteArrayList<>();
```

Comment:
The removal of atomEventListeners seems like a backward step.  I get the comments below on the checker, but this seems like a different thing altogether.

Reply:
The event listener is something I don't think the Radix Engine should manage at this level of abstraction. Much better to do it outside of the api as these callbacks are now effectively called just at the end of public methods. It was originally written to be done for a single atom submission but then morphed into the gargantuan thing it is today.